File: db/java/app/src/main/java/com/example/sales/domain/model/Sales.java
package com.example.sales.domain.model;

import java.time.LocalDateTime;

/**
 * 売上データのEntityクラス
 */
public class Sales {
    private String salesNo;
    private String orderNo;
    private LocalDateTime salesDate;
    private Integer salesDivision;
    private String departmentCode;
    private LocalDateTime startDate;
    private String companyCode;
    private String employeeCode;
    private Integer salesAmount;
    private Integer consumptionTax;
    private String slipComment;
    private Integer redBlackSlipNo;
    private String originalSlipNo;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;

    // デフォルトコンストラクタ
    public Sales() {}

    // Getter/Setter（全フィールド）
    public String getSalesNo() { return salesNo; }
    public void setSalesNo(String salesNo) { this.salesNo = salesNo; }

    public String getOrderNo() { return orderNo; }
    public void setOrderNo(String orderNo) { this.orderNo = orderNo; }

    public LocalDateTime getSalesDate() { return salesDate; }
    public void setSalesDate(LocalDateTime salesDate) { this.salesDate = salesDate; }

    public Integer getSalesDivision() { return salesDivision; }
    public void setSalesDivision(Integer salesDivision) {
        this.salesDivision = salesDivision;
    }

    public String getDepartmentCode() { return departmentCode; }
    public void setDepartmentCode(String departmentCode) {
        this.departmentCode = departmentCode;
    }

    public LocalDateTime getStartDate() { return startDate; }
    public void setStartDate(LocalDateTime startDate) { this.startDate = startDate; }

    public String getCompanyCode() { return companyCode; }
    public void setCompanyCode(String companyCode) { this.companyCode = companyCode; }

    public String getEmployeeCode() { return employeeCode; }
    public void setEmployeeCode(String employeeCode) { this.employeeCode = employeeCode; }

    public Integer getSalesAmount() { return salesAmount; }
    public void setSalesAmount(Integer salesAmount) { this.salesAmount = salesAmount; }

    public Integer getConsumptionTax() { return consumptionTax; }
    public void setConsumptionTax(Integer consumptionTax) {
        this.consumptionTax = consumptionTax;
    }

    public String getSlipComment() { return slipComment; }
    public void setSlipComment(String slipComment) { this.slipComment = slipComment; }

    public Integer getRedBlackSlipNo() { return redBlackSlipNo; }
    public void setRedBlackSlipNo(Integer redBlackSlipNo) {
        this.redBlackSlipNo = redBlackSlipNo;
    }

    public String getOriginalSlipNo() { return originalSlipNo; }
    public void setOriginalSlipNo(String originalSlipNo) {
        this.originalSlipNo = originalSlipNo;
    }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public String getCreatedBy() { return createdBy; }
    public void setCreatedBy(String createdBy) { this.createdBy = createdBy; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public String getUpdatedBy() { return updatedBy; }
    public void setUpdatedBy(String updatedBy) { this.updatedBy = updatedBy; }
}

File: db/go/sales_management/internal/model/sales.go
package model

import (
	"database/sql"
	"time"
)

// Sales 売上のドメインモデル（ヘッダー）
type Sales struct {
	SalesNo        string         `db:"売上番号"`
	SalesDate      sql.NullTime   `db:"売上日"`
	SalesCategory  int            `db:"売上区分"` // 1:通常売上、2:返品、3:値引
	OrderNo        sql.NullString `db:"受注番号"`
	DepartmentCode string         `db:"部門コード"`
	StartDate      time.Time      `db:"開始日"`
	CompanyCode    string         `db:"取引先コード"`
	SalesAmount    int            `db:"売上金額"`
	ConsumptionTax int            `db:"消費税"`
	CorrectionNo   sql.NullString `db:"訂正番号"`  // 赤黒伝票の黒伝票番号
	OriginalSlipNo sql.NullString `db:"元伝票番号"` // 赤黒伝票の元伝票番号
	CreatedAt      time.Time      `db:"作成日時"`
	CreatedBy      string         `db:"作成者名"`
	UpdatedAt      time.Time      `db:"更新日時"`
	UpdatedBy      string         `db:"更新者名"`
}

// SalesDetail 売上明細のドメインモデル
type SalesDetail struct {
	SalesNo          string         `db:"売上番号"`
	DetailNo         int            `db:"明細番号"`
	ProductCode      string         `db:"商品コード"`
	ProductName      sql.NullString `db:"商品名"`
	SellingPrice     int            `db:"販売単価"`
	ShippedQuantity  int            `db:"出荷済数量"`
	Quantity         int            `db:"数量"`
	DiscountAmount   int            `db:"値引額"`
	BillingDate      sql.NullTime   `db:"請求日"`
	InvoiceNo        sql.NullString `db:"請求番号"`
	BillingDelayFlag sql.NullInt32  `db:"請求遅延区分"`
	AutoJournalDate  sql.NullTime   `db:"自動仕訳日"`
	CreatedAt        time.Time      `db:"作成日時"`
	CreatedBy        string         `db:"作成者名"`
	UpdatedAt        time.Time      `db:"更新日時"`
	UpdatedBy        string         `db:"更新者名"`
}
